### DISCUSSION TITLE
qemu

### DISCUSSION THREAD

**Discussion by fgsoftware1** [fgsoftware1/wolfieOS#64] [General]
i can't boot system on qemu

**Comment by circutrider21** [comment]
What happens when you try to boot

**Reply by fgsoftware1** [reply]
> What happens when you try to boot

qemu could not read from boot disk

**Reply by circutrider21** [reply]
what args did you use.
By args I mean this...
````
qemu-system-i386 <args>
````

**Reply by fgsoftware1** [reply]
> what args did you use.
> By args I mean this...
> 
> ```
> qemu-system-i386 <args>
> ```

i use -cdrom and i use qemu-system-x86_64

**Reply by circutrider21** [reply]
type the full command you typed into the terminal here

**Reply by fgsoftware1** [reply]
qemu-system-x86_64 -cdrom build/fgOS.iso

**Reply by fgsoftware1** [reply]
> type the full command you typed into the terminal here

build/fgOS.iso file is generated on build process i don't push it to github to minify space usage

**Reply by circutrider21** [reply]
dosen't qemu give a message along the lines of
````
Could not read from Cdrom (Code xxxx)
````

**Reply by fgsoftware1** [reply]
> dosen't qemu give a message along the lines of
> 
> ```
> Could not read from Cdrom (Code xxxx)
> ```

no.
only said "could not read from boot disk" on both hard disk,floppy and DVD/CD

**Reply by circutrider21** [reply]
hmm

**Reply by circutrider21** [reply]
could you upload the iso by any chance

**Reply by fgsoftware1** [reply]
> could you upload the iso by any chance

yes.
wait a moment

**Reply by fgsoftware1** [reply]
> could you upload the iso by any chance

https://github.com/fgsoftware1/fgOS/releases/tag/v1.0a

**Reply by circutrider21** [reply]
Well, on my computer qemu gives me code 0004. This means that the iso is not bootable.

**Reply by circutrider21** [reply]
upon further inspection, I noticed that grub wasn't installed to the iso (the folder /boot/grub/roms was empty)

**Reply by circutrider21** [reply]
you should try installing grub to the iso again

**Reply by fgsoftware1** [reply]
> upon further inspection, I noticed that grub wasn't installed to the iso (the folder /boot/grub/roms was empty)

i notice that too but i don't know how solve

**Reply by fgsoftware1** [reply]
> you should try installing grub to the iso again

how?

**Reply by circutrider21** [reply]
````
grub-mkrescue -o fgOS.iso <folder to where iso files (like grub.cfg, fgOS.bin) are stored>
````

**Reply by fgsoftware1** [reply]
> ```
> grub-mkrescue -o fgOS.iso <folder to where iso files (like grub.cfg, fgOS.bin) are stored>
> ```

same error

**Reply by fgsoftware1** [reply]
> ```
> grub-mkrescue -o fgOS.iso <folder to where iso files (like grub.cfg, fgOS.bin) are stored>
> ```

bin file is generated trough linking objects right?

**Reply by circutrider21** [reply]
yes

**Reply by fgsoftware1** [reply]
> yes

take a look at grub-mkrescue output
`Drive current: -outdev 'stdio:build/fgOS.iso'
Media current: stdio file, overwriteable
Media status : is blank
Media summary: 0 sessions, 0 data blocks, 0 data, 68.0g free
Added to ISO image: directory '/'='/tmp/grub.XzI8XU'
xorriso : UPDATE :       3 files added in 1 seconds
Added to ISO image: directory '/'='/home/admin/Desktop/projects/fgOS/isodir'
xorriso : UPDATE :       6 files added in 1 seconds
ISO image produced: 184 sectors
Written to medium : 184 sectors at LBA 0
Writing to 'stdio:build/fgOS.iso' completed successfully.`

**Reply by circutrider21** [reply]
Its only adding 6 files

**Reply by circutrider21** [reply]
grub itself is ~200 files

**Reply by circutrider21** [reply]
so I can tell there is no grub in the iso
This is why you should use [limine](https://github.com/limine-bootloader/limine)

**Reply by fgsoftware1** [reply]
> so I can tell there is no grub in the iso
> This is why you should use [limine](https://github.com/limine-bootloader/limine)

i found this:

As it turns out, this happens when the system you're running this command on has no (accessible) /boot/grub directory. It will silently fail and produce an empty ISO file.

When running the same command on a normal desktop installation of Linux, the command produces a working bootable image.

**Reply by circutrider21** [reply]
Wait, are you using windows subsystem for linux

**Reply by fgsoftware1** [reply]
> Wait, are you using windows subsystem for linux

no im compiling on docker.
forget this i solve this.
now i got another error:

**error: file /boot/fgOS.bin not found**
**error: you need to load kernel first**
**press any key to continue...**

**Reply by fgsoftware1** [reply]
if you want take a look at iso:
[iso](https://github.com/fgsoftware1/fgOS/releases/tag/v1.0a)

**Reply by fgsoftware1** [reply]
solved
